mesonbuild/build.py

```diff
@@ -1718,6 +1728,7 @@ class GeneratedList(HoldableObject):
     generator: Generator
     subdir: str
     preserve_path_from: T.Optional[str]
+    depend_files: T.List[File]
```

Comment:
```suggestion
    depend_files: T.List[File] = dataclasses.field(default_factory=list)
```
Otherwise we're lying to the type system by telling it that this will be a list, then setting it to None, then fixing it in the `__post_init__`.